x/foundation/keeper/grant.go

```diff
@@ -57,23 +55,21 @@ func (k Keeper) GetAuthorization(ctx sdk.Context, granter string, grantee sdk.Ac
 
 	var auth foundation.Authorization
 	if err := k.cdc.UnmarshalInterface(bz, &auth); err != nil {
-		return nil, err
+		panic(err)
 	}
 
 	return auth, nil
 }
 
-func (k Keeper) setAuthorization(ctx sdk.Context, granter string, grantee sdk.AccAddress, authorization foundation.Authorization) error {
+func (k Keeper) setAuthorization(ctx sdk.Context, granter string, grantee sdk.AccAddress, authorization foundation.Authorization) {
 	store := ctx.KVStore(k.storeKey)
 	key := grantKey(grantee, authorization.MsgTypeURL(), granter)
 
 	bz, err := k.cdc.MarshalInterface(authorization)
 	if err != nil {
-		return err
+		panic(err)
```

Comment:
Is this `panic` no problem? Because this `setAutorization` function is call in `Grant` function of `foundation` Msg.

Reply:
No problem. The server won't panic, and it just converts it to the error. You can also see them on many other places (e.g. functions like `MustMarshal()` actually panics inside it).